src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -7551,6 +7640,14 @@ address generate_avx_ghash_processBlocks() {
         StubRoutines::_cipherBlockChaining_decryptAESCrypt = generate_cipherBlockChaining_decryptAESCrypt_Parallel();
       }
     }
+    if (UseAESIntrinsics) {
+      if (VM_Version::supports_avx512_vaes() &&  VM_Version::supports_avx512vl() && VM_Version::supports_avx512dq()) {
```

Comment:
Why duplicate already existing checks? Move code there and add comment for which intrinsic code is generated.